src/main/java/seedu/address/logic/commands/AddTransactionCommand.java

```diff
@@ -0,0 +1,80 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_COST;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_DESCRIPTION;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NAME;
+
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.transaction.Transaction;
+
+
+/**
+ * Adds a transaction to the address book.
+ */
+public class AddTransactionCommand extends Command {
+
+    public static final String COMMAND_WORD = "addTransaction";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Adds a transaction to the address book. "
+            + "Parameters: "
+            + PREFIX_NAME + "NAME "
+            + PREFIX_COST + "COST "
+            + PREFIX_DESCRIPTION + "DESCRIPTION "
+            + "Example: " + COMMAND_WORD + " "
+            + PREFIX_NAME + "John Doe "
+            + PREFIX_COST + "25.00 "
+            + PREFIX_DESCRIPTION + "bread";
+
+    public static final String MESSAGE_SUCCESS = "New transaction added: %1$s";
+
+    public static final String MESSAGE_DUPLICATE_TRANSACTION = "This transaction already exists in the address book";
```

Comment:
Are we enforcing uniqueness of transactions?

Reply:
IIRC yes